core/common.ts

```diff
@@ -86,38 +91,58 @@ export function setMainWorkspace(workspace: Workspace) {
 }
 
 /**
- * Currently selected copyable object.
+ * Returns the current selection.
  */
-let selected: ISelectable | null = null;
+export function getSelected(): ISelectable | null {
+  const focused = getFocusManager().getFocusedNode();
```

Comment:
I am curious about how well this will work. 

One scenario is that in context menu items sometimes the callback would use getSelected, because the block is still selected when the menu is open. But the block won’t be focused when the menu is open. So we would need to call this out as a breaking change. I am curious if there are other situations where a block would have been selected but not (active) focused and I’m not sure I can think of any, maybe in some widget div scenarios

Reply:
I think the scenario you describe would be broken as of #8938 not this change, but it's equally relevant. It's definitely the case now that right clicking will not maintain block selection since the selection highlight is cleared on focus loss.

It's something worth discussing, I think. Does something really have selection if it doesn't have input focus? My assumption is no which is why the change moved in this direction. If the answer should be yes, then we need to approach the auto-selection differently.

Reply:
I think logically it's fine, but it is a big behavior change that the block isn't selected anymore while you're interacting with non-Blockly parts of the UI (e.g. context menu or dev console) so we just need to be super clear about that in our communication.

Reply:
What's the best way for me to indicate that bit (per being super clear)?

Reply:
Update the PR description of this PR (and/or any others you've made that are breaking) to include a `## Breaking Changes` section that explains 1) who is broken by the change 2) what they need to do in response to the breaking change.

So for the issue about the block not being selected anymore, provide workarounds like using the (now passively-) focused block instead of selected block, using the scope for context menu items and shortcuts instead of selected, etc.

For this PR, you might need other details like what new methods need to be implemented if any for bubbles, etc.

Then remind me to call this out in the handwritten release notes we'll attach to the github release/forum announcement as well.